pkg/tdexdconnect/tdexdconnect.go

```diff
@@ -17,6 +17,10 @@ func Encode(
 		return nil, errors.New("http protocol invalid with cert provided")
 	}
 
+	if len(certBytes) == 0 && proto == "https" {
+		return nil, errors.New("https protocol invalid without cert provided")
+	}
+
```

Comment:
Why? This should be allowed indeed. In case proto is https and there's no cert, then CA cert (`credentials.NewTls(nil)`) should be used.

Reply:
Then [this](https://github.com/sekulicd/tdex-daemon/blob/967d788e7f341319a58a65a9941cdd577f745d40/internal/interfaces/http/tdex_connect_handler.go#L66) is invalid as well no?

Reply:
yes. In general, I thought e10bb6b49280d48b71198c23d2a19a3002b9249e was ok. What are later changes for?

Reply:
we cant use credentials.NewTls(nil) for self signed cert since "...not trusted authority..." error is thrown.

Reply:
@altafan checks removed.